src/plugins/data/common/exports/export_csv.tsx

```diff
@@ -10,40 +10,26 @@
 
 import { FormatFactory } from 'src/plugins/data/common/field_formats/utils';
 import { Datatable } from 'src/plugins/expressions';
+import { createEscapeValue } from './escape_value';
 
 export const LINE_FEED_CHARACTER = '\r\n';
-const nonAlphaNumRE = /[^a-zA-Z0-9]/;
-const allDoubleQuoteRE = /"/g;
 export const CSV_MIME_TYPE = 'text/plain;charset=utf-8';
 
-// TODO: enhance this later on
-function escape(val: object | string, quoteValues: boolean) {
-  if (val != null && typeof val === 'object') {
-    val = val.valueOf();
-  }
-
-  val = String(val);
-
-  if (quoteValues && nonAlphaNumRE.test(val)) {
-    val = `"${val.replace(allDoubleQuoteRE, '""')}"`;
-  }
-
-  return val;
-}
-
 interface CSVOptions {
   csvSeparator: string;
   quoteValues: boolean;
+  escapeFormulaValues?: boolean;
   formatFactory: FormatFactory;
   raw?: boolean;
 }
 
 export function datatableToCSV(
   { columns, rows }: Datatable,
-  { csvSeparator, quoteValues, formatFactory, raw }: CSVOptions
+  { csvSeparator, quoteValues, formatFactory, raw, escapeFormulaValues = false }: CSVOptions
```

Comment:
Can we make this a required argument instead? I found it hard to trace the code that is using the default value here vs actually providing a value. It looks like only the Reporting integration has a way to escape formulas, with the `xpack.reporting.csv.escapeFormulaValues` setting, and I was expecting that other places in the code would also use escaping.